pkg/sql/sqlstats/insights/integration/integration_test: TestInsightsIntegrationForContention failed

pkg/sql/sqlstats/insights/integration/integration_test.TestInsightsIntegrationForContention [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13944932?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13944932?buildTab=artifacts#/) on release-23.1 @ [9b3c1cabbcbfb491d33b6ea8943c0762a17892b6](https://github.com/cockroachdb/cockroach/commits/9b3c1cabbcbfb491d33b6ea8943c0762a17892b6):


```
=== RUN   TestInsightsIntegrationForContention
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f75d8af2c2b6af04e0405b19fd4d23f3/logTestInsightsIntegrationForContention554266633
    test_log_scope.go:79: use -show-logs to present logs inline
    insights_test.go:600: condition failed to evaluate within 45s: waitingTxnFingerprintID is default value
        query: UPDATE t SET s = '_' WHERE id = '_', totalContentionFromQueryMs: 0.725134, contentionFromEventMs: 0.725134,
        				"schemaName: public, dbName: defaultdb, tableName: t, indexName: t_pkey, waitingTxnFingerprintID: 0000000000000000
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f75d8af2c2b6af04e0405b19fd4d23f3/logTestInsightsIntegrationForContention554266633
--- FAIL: TestInsightsIntegrationForContention (51.21s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInsightsIntegrationForContention.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36021

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -760,12 +760,13 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 	// lookup this id will result in the resolver potentially missing the event.
 	txnIDCache := tc.ApplicationLayer(0).SQLServer().(*sql.Server).GetTxnIDCache()
 	txnIDCache.DrainWriteBuffer()
+	var expectedWaitingTxnFingerprintID appstatspb.TransactionFingerprintID
 	testutils.SucceedsSoon(t, func() error {
 		waitingTxnFingerprintID, ok := txnIDCache.Lookup(waitingTxnID)
 		if !ok || waitingTxnFingerprintID == appstatspb.InvalidTransactionFingerprintID {
 			return fmt.Errorf("waiting txn fingerprint not found in cache")
 		}
-
+		expectedWaitingTxnFingerprintID = waitingTxnFingerprintID
 		return nil
 	})
 
@@ -794,11 +795,12 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 		// at least 1 row matches the one we're looking for.
 		foundRow := false
 		var lastErr error
+		rowsCount := 0
 		for rows.Next() {
 			if err != nil {
 				return err
 			}
-
+			rowsCount++
 			var totalContentionFromQueryMs, contentionFromEventMs float64
 			var queryText, schemaName, dbName, tableName, indexName, waitingTxnFingerprintID string
 			err = rows.Scan(&queryText, &totalContentionFromQueryMs, &contentionFromEventMs, &schemaName, &dbName, &tableName, &indexName, &waitingTxnFingerprintID)
@@ -842,7 +844,7 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 			}
 
 			if waitingTxnFingerprintID == "0000000000000000" || waitingTxnFingerprintID == "" {
-				lastErr = fmt.Errorf("waitingTxnFingerprintID is default value\n%s", prettyPrintRow)
+				lastErr = fmt.Errorf("expected waitingTxnFingerprintID to be %d, but got %s. \nScanned row: \n%s", expectedWaitingTxnFingerprintID, waitingTxnFingerprintID, prettyPrintRow)
 				continue
 			}
 
@@ -851,6 +853,8 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 		}
 
 		if !foundRow && lastErr != nil {
+			t.Logf("rowsCount = %d", rowsCount)
+			t.Logf("ContentionRegistry: \n%s", tc.ApplicationLayer(0).ExecutorConfig().(sql.ExecutorConfig).ContentionRegistry.String())
 			return lastErr
 		}
 

